### PR TITLE
display name changed as per site settings

### PR DESCRIPTION
**Pull request by Ahmedgagan**

### CODE REVIEW

File: assets/javascripts/discourse/components/group-assignments-filter.js.es6 [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
+  displayName: "",
+
+  init(){
+    if(this.siteSettings.prioritize_username_in_ux){
+      this.set("displayName",this.filter.username);
+    }else{
+      this.set("displayName",this.filter.displayName);
+    }
+    this._super(...arguments);
+  }
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/components/group-assignments-filter.js.es6:14]: I think a better approach here would be to use a computed property. If you search core for `@discourseComputed` you'll find some examples. [For example](https://github.com/discourse/discourse/blob/fbf29443bdb84e0a07a34a82d7893ab967854bd9/app/assets/javascripts/admin/components/admin-report-table-cell.js#L11-L14).

Comment (Ahmedgagan) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: Solved this in new commit

Comment (davidtaylorhq) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: This should be computed based on all the properties it uses. So `filter.username` and `filter.displayName` should also be included in the `discourseComputed(...)`

Comment (Ahmedgagan) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: Sure, just added these too in discourseComputed() in latest commit

Comment (davidtaylorhq) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: And then we can pass them as variables into the function, just like you did for the site setting :)

Comment (Ahmedgagan) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: Oooh sorry for that😅😅

Comment (Ahmedgagan) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:13]: Updated this one😄

File: assets/javascripts/discourse/components/group-assignments-filter.js.es6 [hunk 2]
```diff
@@ -1,5 +1,14 @@
+import discourseComputed from "discourse-common/utils/decorators";
 import Component from "@ember/component";
 
 export default Component.extend({
-  tagName: "li"
+  tagName: "li",
+
+  @discourseComputed("siteSettings.prioritize_username_in_ux", "filter.username", "filter.displayName")
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/components/group-assignments-filter.js.es6:7]: Where does `filter.displayName` come from? Should that be `filter.name`?

Apart from that question, this is looking great :D

Comment (Ahmedgagan) [reply assets/javascripts/discourse/components/group-assignments-filter.js.es6:7]: filter.name property doesn't value when the user doesn't set name property in the signup, but displayName always have the value

**Pull request merged**